src/note.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  SolidDataset, Thing,
+  getThing, createThing,
+  addUrl,
+  setInteger, setDecimal, setStringNoLocale, setBoolean,
+  getStringNoLocale, getInteger, getBoolean, getUrl,
+} from '@inrupt/solid-client';
+import { createNS } from "@ontologies/core"
+
+import {arrayToThings, thingsToArray} from "./collections"
+
+const noteNSUrl = "https://mysilio.garden/ontologies/note#"
+const noteNS = createNS(noteNSUrl)
+
+function addKeyValToThing(thing: Thing, key: string, value: any, path: number[]): Thing[] {
+  if (Array.isArray(value)) {
+    const [arrayThing, ...restOfThings] = arrayToThings(value, createThingFromSlateJSOElement, [...path, 0])
+    if (arrayThing) {
+      return [
+        addUrl(thing, key, arrayThing),
+        arrayThing,
+        ...restOfThings
+      ]
+    } else {
+      // TODO: is this the right thing to do? arrayThing should never be falsy so maybe throw an error?
+      return []
+    }
+  } else {
+
+    switch (typeof value) {
+      case 'string':
+        return [setStringNoLocale(thing, key, value)]
+      case 'boolean':
+        return [setBoolean(thing, key, value)]
+      case 'number':
+        if (Number.isInteger(value)) {
+          return [setInteger(thing, key, value)]
+        } else {
+          return [setDecimal(thing, key, value)]
+        }
+      default:
+        return [thing]
+    }
+  }
+
+}
+
+export function createThingFromSlateJSOElement(o: any, path: number[]): Thing[] {
+  const thing = createThing({ name: (o.id ? `${o.id}` : `el-${path.join("-")}`) })
+  const otherThings = Object.keys(o).reduce((m: Thing[], k: string) => {
+    const [mThing, ...mRest] = m;
+    const [thing, ...restOfThings] = addKeyValToThing(mThing, noteNS(k).value, o[k], path)
+    return [thing, ...mRest, ...restOfThings]
+  }, [thing])
+  return [thing, ...otherThings]
+}
+
+function childrenArrayFromDataset(dataset: SolidDataset, childrenUrl: string) {
+  const thing = getThing(dataset, childrenUrl);
+
+  if (thing) {
+    return thingsToArray(thing, dataset, noteThingToSlateObject)
+  } else {
+    return []
+  }
+}
+
+const childrenPred = noteNS('children')
+export function noteThingToSlateObject(thing: Thing, dataset: SolidDataset) {
+  const obj: any = {}
+  for (const pred in thing.predicates) {
+    const [, key] = pred.split(noteNSUrl)
+    if (key) {
+      if (key === 'children') {
```

Comment:
Why not instead test to see if `pred === childrenPred.value` or `pred === noteNS('children').value`?

To me that reads a little cleaner than splitting on the predicate namespace.